arm/examples/sdk.go

```diff
@@ -0,0 +1,168 @@
+package examples
+
+// Copyright 2017 Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import (
+	"fmt"
+	"os"
+
+	"bytes"
+	"encoding/json"
+	"github.com/Azure/azure-sdk-for-go/arm/examples/helpers"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// SDK represents a working authenticated SDK structure.
+type SDK struct {
```

Comment:
I'm not convinced that this type and `Authentication` are both needed. Though, I do like the idea of consolidating all of this information, it makes for cleaner code.

How would you feel about removing the `SDK` type, making the Constructor `NewSDK` instead be a constructor for an `Authentication`. You could then have `EnsureResourceGroup` take an Authentication as its first parameter. 

I'm open to other ideas as well.